plugins/uefi/fu-uefi-bootmgr.c

```diff
@@ -325,7 +325,19 @@ fu_uefi_bootmgr_bootnext (const gchar *esp_path,
 	shim_app = fu_uefi_get_esp_app_path (esp_path, "shim", error);
 	if (shim_app == NULL)
 		return FALSE;
-	if (!g_file_test (shim_app, G_FILE_TEST_EXISTS)) {
+	if (g_file_test (shim_app, G_FILE_TEST_EXISTS)) {
+		/* use a custom copy of shim for firmware updates */
+		if (flags & FU_UEFI_BOOTMGR_FLAG_USE_SHIM_UNIQUE) {
+			shim_cpy = fu_uefi_get_esp_app_path (esp_path, "shimfwupd", error);
+			if (shim_cpy == NULL)
+				return FALSE;
+			if (!fu_uefi_copy_asset (shim_app, shim_cpy, error))
```

Comment:
I think you're missing a check for `fu_uefi_cmp_asset` on this file

Reply:
Ahh, to avoid copying if the files are the same?

Reply:
I wonder if `fu_uefi_copy_asset()` should always do the `fu_uefi_cmp_asset()` step.

Reply:
Yeah to avoid copying if the same.

Reply:
Fixed, thanks. I opted not to always do the _cmp_ to allow debugging this later.